### PR TITLE
refactor - Internal state of cell

### PR DESCRIPTION
**Pull request by Dylanflini**
# Summary

## References

# Self review

- [ ] I have reviewed my PR changes and left comments/questions when necessary
- [ ] I have added a screenshot for the applied changes.
- [ ] I have followed the style guidelines of this project.
- [ ] I have added and ran the tests locally.
- [ ] I have ran the test coverage reports locally.
- [ ] I have made corresponding changes to the documentation (README and API).
- [ ] I have added the abstraction layers needed to separate responsibilities.

## Impact of these changes

This changes through `Component` impacts:

- `Component1`
- `Component2`
- `Component3`

## Evidences of expected changes

Add evidences here about each impact generated by your changes

# Type of change

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Hot fix (fixes an issue in master branch)
- [ ] Docs (Documentation only changes)
- [ ] Style (Changes that do not affect the meaning of the code, formatting, missing semi-colons, etc)
- [ ] Refactor (Changes that neither fixes a bug or adds a feature)
- [ ] Performance (Changes that improve performance)
- [ ] Chore (Changes to the build process or auxiliary tools and libraries)

# Squash and merge

We are following [Angular commit messages guidelines](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#-git-commit-guidelines)

```
<type>(<scope>): <Jira-Issue subject>
<BLANK LINE>
<optional body>
<BLANK LINE>
<optional footer>
```

Types

```
feat: A new feature
fix: A bug fix
docs: Documentation only changes
style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
refactor: A code change that neither fixes a bug nor adds a feature
perf: A code change that improves performance
test: Adding missing or correcting existing tests
chore: Changes to the build process or auxiliary tools and libraries such as documentation generation
```

Make sure you squash and merge your feature/bugfix branches into develop/release/hotfix branches using the suggested format.

Some examples.

```
feat: DP-1234 Edit assessment information
Frontend changes for editing an assessment:
- improve assign modal to receive an existing assessment
- integrate the reasign end point.
- couple of warnings removed
BREAKING CHANGE: the assign modal now receives 3 required parameters
```

More examples [here](https://www.conventionalcommits.org/en/v1.0.0-beta.2/#summary)

### CODE REVIEW

File: minesweeper/domain/models/Cell.model.ts [hunk 1]
```diff
@@ -1,5 +1,16 @@
 import { IPosition } from '@minesweeper/domain/commons.type';
 
 export interface CellModel extends IPosition {
-  exposed: boolean;
+  /**
+   * represent internal state of cell
+   *
+   * if is "undefined" it means that the cell is not exposed and the user doesn't know if this cell is a bomb or not
+   */
+  isBomb?: boolean;
```
Comment (Dylanflini) [inline minesweeper/domain/models/Cell.model.ts:9]: I changed idea of "state" on cell Model for property "isBomb", i think this way is more intuitive

Comment (danielglobant) [reply minesweeper/domain/models/Cell.model.ts:9]: I prefer the word hasBomb instead of isBomb, I think it is more clear

Comment (Dylanflini) [reply minesweeper/domain/models/Cell.model.ts:9]: Okaaa

File: src/minesweeper/use-cases/markFlag/markFlag.validations.ts [hunk 2]
```diff
@@ -13,15 +18,14 @@ export enum MarkFlagUCError {
   NO_FLAGS_AVAILABLE = "[Error] You can't add a flag when you there is no flags available",
 }
 
-export const makeValidations = (element: IMarkFlagProps, board: BoardModel) => {
-  const [columnMarked, rowMarked] = element.position;
-
+export const makeValidations = (props: IMarkFlagProps, board: BoardModel) => {
+  const [columnMarked, rowMarked] = props.position;
+  const samePosition = ({ position: [column, row] }: IPosition) =>
```
Comment (danielglobant) [inline src/minesweeper/use-cases/markFlag/markFlag.validations.ts]: this should use the util created

**Review by danielglobant — APPROVED** [review]

**Pull request merged**